openmdao/docs/pre_tag.py

```diff
@@ -0,0 +1,94 @@
+#pre_tag.py
+#A script that finds occurrences of the .. tags:: directive
+#and sets up the structure of the tags directory.  One file
+#is created for each subject tag, that file contains links to
+#each instance of the tag throughout the docs.
+
+import sys, os, shutil, re
+
+def make_tagdir():
+    #clean up tagdir, create tagdir, return tagdir
+    dir = os.path.dirname(__file__)
+    tagdir = os.path.join(dir, "tags")
+
+    if os.path.isdir(tagdir):
+      shutil.rmtree(tagdir)
+
+    if not os.path.isdir(tagdir):
+       os.mkdir(tagdir)
+
+    return tagdir
+
+def make_tagfiles(docdirs, tagdir):
+    #pull tags from each file, then make a file
+    #for each tag, containing all links to tagged files.
+    for docdir in docdirs:
+        for dirpath, dirnames, filenames in os.walk(docdir):
+            for filename in filenames:
+                #the path to the file being read for tags
+                sourcefile = os.path.join(dirpath, filename)
+                #a file object for the file being read for tags
+                textfile = open( sourcefile, 'r')
```

Comment:
use a 'with open(sourcefile, 'r') as textfile:'  block around your 'filetext=textfile.read()' and you can remove the textfile.close()  line.   It's a little shorter and will always close the file even if there's an exception during the read.